boot/seal.go

```diff
@@ -149,6 +151,8 @@ func sealKeyToModeenvImpl(key, saveKey keys.EncryptionKey, model *asserts.Model,
 		return sealKeyToModeenvUsingFDESetupHook(key, saveKey, model, modeenv, flags)
 	}
 
+	relock := flags.Unlocker()
+	defer relock()
```

Comment:
the contract I chose in https://github.com/snapcore/snapd/pull/13185 is that it's ok for unlocker to be passed as nil and then we don't call it, as it will not be used in a lot of places I think it's ok